src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -46,38 +46,25 @@ function useDecline () {
   })
 }
 
-function useFetchCounterparties () {
+function useTransactionsWithCounterparties () {
```

Comment:
I changed the way this worked actually back to how you were originally doing it.

That's because the update caching method wasn't actually working how we thought. Remember how

`updatedTransactions = transactions.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })`

worked to update the transactions, but

`updatedTransactions = transactions.map(transaction => ({
  ...transaction,
  counterparty: agent
})`

didn't?

That's because we were actually relying on the *sideeffect* of `Object.assign`, not the return value. We were modifying the objects in cache directly, not using writeCache. We don't want to do this because it's unspecified behavior. So I switched to the simpler thing of just returning updated versions of the transaction lists.

(none of this was directly to do with the bug, just ended up cleaning this up while fixing the bug).

Reply:
Interesting thought.  Recently, when reading through the Apollo docs to consider best solutions for PT Ticket #1844 and found that Apollo recommends the use of 'Object.assign' when using Subscriptions' `updateQuery` option to update cached data from Queries.  I'd love to contrast these approaches together to understand how the `updateQuery` option handles cache writing differently, such that the use 'Object.assign' is a preferred implementation.

Nonetheless, good call to avoid any side effects. :+1: